quartz/plugins/transformers/roam.ts

```diff
@@ -0,0 +1,132 @@
+          const dropdown = `<select class="roam-or-component">${options.map((option: string) => `<option>${option}</option>`).join("")}</select>`
+          return dropdown
+        })
+      }
+      if (opts.TODOComponent) {
+        src = src.toString()
+        src = src.replaceAll(TODORegex, (value, ...capture) => {
+          const checkbox = '<input type="checkbox" disabled>'
+          return checkbox
+        })
+      }
+      if (opts.DONEComponent) {
+        src = src.toString()
+        src = src.replaceAll(DONERegex, (value, ...capture) => {
+          const checkbox = '<input type="checkbox" checked disabled>'
+          return checkbox
+        })
+      }
+      if (opts.videoComponent) {
+        //youtube first then regular video links
+        src = src.toString()
+        src = src.replaceAll(youtubeRegex, (value, ...capture) => {
+          const [match, text] = capture
+          const video = `<iframe width="560" height="315" src="https://www.youtube.com/embed/${text}" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>`
+          return video
+        })
+        src = src.replaceAll(videoRegex, (value, ...capture) => {
+          const [match, text] = capture
+          const video = `<video controls><source src="{${match}}" type="video/mp4"><source src="${match}" type="video/webm"></video>`
+          return video
+        })
+      }
+      if (opts.audioComponent) {
+        src = src.toString()
+        src = src.replaceAll(audioRegex, (value, ...capture) => {
+          const [match, text] = capture
+          const audio = `<audio controls>
+            <source src="${match}" type="audio/mpeg">
+            <source src="${match}" type="audio/ogg">
+            Your browser does not support the audio tag.
+          </audio>`
+          return audio
+        })
+      }
+      if (opts.pdfComponent) {
+        src = src.toString()
+        src = src.replaceAll(pdfRegex, (value, ...capture) => {
+          const [match, text] = capture
+          const pdf = `<embed src="${match}" type="application/pdf" width="600" height="800">`
+          return pdf
+        })
+      }
+      if (opts.blockquoteComponent) {
+        src = src.toString()
+        src = src.replaceAll(blockquoteRegex, (value, ...capture) => {
+          const bq = `>`
+          return bq
+        })
+      }
+      // TODO attributes in roam are sort of like block level frontmatter or tags
+
+      src = src.toString()
+      src = src.replaceAll(roamHighlightRegex, (value, ...capture) => {
+        const [match, text] = capture
+        const highlight = `==${match}==`
+        return highlight
+      })
+      // roam italics
+      src = src.replaceAll(roamItalicRegex, (value, ...capture) => {
+        const [match, text] = capture
+        const italic = `*${match}*`
+        return italic
+      })
```

Comment:
Probably better to do this in markdownPlugins imo

Reply:
I have not had any luck converting the components like audioComponent to a marketing plugin. Any pointers/ guidance / suggestions? Kind of stuck

Reply:
we use this https://github.com/syntax-tree/mdast-util-find-and-replace

try that out?

Reply:
@jackyzha0 just pushed a WIP to show the issue. Maybe there's an issue with my regex but on LN 192 my audio component matches the component `{{[[audio]]: someText }}` but not one with a url like this `{{[[audio]]: www.google.com }}`?

Reply:
ah gfm autolinks anything that looks like a link, you'll either have to move your plugin before it or parse out the link yourself

Reply:
Isn't it already before GFM though?
<img width="694" alt="image" src="https://github.com/jackyzha0/quartz/assets/4028391/132d7a98-3f13-4117-af0f-996c7369b113">

Reply:
Also interesting because as a `textTransform` the regex has no issue but as a `markdownPlugin` it fails.